github-account

milabozhevskaya opened the pull request.

milabozhevskaya committed:
chore: set environment

milabozhevskaya committed:
feat: add shared components

milabozhevskaya committed:
feat: add entities components

milabozhevskaya committed:
feat: add features components

milabozhevskaya committed:
feat: add widgets components

milabozhevskaya committed:
feat: add pages components

milabozhevskaya committed:
feat: add app component

remadyt commented on src/app/app.tsx:
```diff
@@ -0,0 +1,19 @@
+import { useService } from "@/shared/api/useService";
+import { getUser } from "@/shared/api/getUser";
+
+const App = (): JSX.Element => {
```
не особо вижу смысла в JSX.Element, тс и сам понимает возвращаемо значение

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/9e0874b4113141c1a5f773195822c97595d3120f

remadyt commented on src/entities/repo/ui/card/index.tsx:
```diff
@@ -0,0 +1,14 @@
+import css from "./styles.module.scss";
+
+const Card = ({ name, description, htmlURL }: Repo): JSX.Element => (
+	<div className={css.card}>
```
советую ставить отступ в 2 пробела, вместо 4

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/d94a411ba36c617de639c1efec64f04cc87f914e

remadyt commented on src/entities/repo/ui/card/index.tsx:
```diff
@@ -0,0 +1,14 @@
+import type { Repo } from "@/shared/types";
+
+import css from "./styles.module.scss";
```
лучше styles/classes

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/9e0874b4113141c1a5f773195822c97595d3120f

remadyt commented on src/entities/user/index.tsx:
```diff
@@ -0,0 +1,42 @@
+	following,
+}: Exclude<User, "publicRepos">): JSX.Element => (
+	<div className={css["user-info"]}>
+		<Avatar>
```
не совсем понятна эта обретка

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/9e0874b4113141c1a5f773195822c97595d3120f

remadyt commented on src/entities/user/styles.module.scss:
```diff
@@ -0,0 +1,64 @@
+	flex-direction: column;
+	align-items: center;
+	row-gap: 16px;
+	font-family: Inter, sans-serif;
```
выглядит как глобальный шрифт

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c2d2f682c9fa3798af5898470bebe96248bcfd2b

remadyt commented on src/entities/user/styles.module.scss:
```diff
@@ -0,0 +1,64 @@
+	}
+
+	&__login {
+		font-weight: 400;
```
это не по дефолту? проверь все места

remadyt commented on src/entities/user/styles.module.scss:
```diff
@@ -0,0 +1,64 @@
+	&__login {
+		font-weight: 400;
+		font-size: 18px;
+		color: #0064eb;
```
вынеси в переменную

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c2d2f682c9fa3798af5898470bebe96248bcfd2b

remadyt commented on src/features/search/index.tsx:
```diff
@@ -0,0 +1,16 @@
+	as: Component = "input",
+	onSearch,
+	...props
+}: {
```
вынеси пропсы отдельно,пока сложно понять что тут написано

remadyt commented on src/pages/empty-page/styles.module.scss:
```diff
@@ -0,0 +1,31 @@
+		svg {
+			width: 100%;
+			height: 100%;
+			fill: #808080;
```
все цвета в переменные

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/aeade1323e92f346b73ee1880da5ec9ce5cf1166

remadyt commented on src/shared/api/getUser.ts:
```diff
@@ -0,0 +1,31 @@
+import type { User } from "../types";
+
+const getUser = async ({ query }: { query: string }): Promise<User> => {
+  const response = await fetch(`https://api.github.com/users/${query}`);
```
вынеси в env и в константу https://api.github.com/

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/52a89fa0dd8959a63572ce610bff9a3cdb8dd7b2

remadyt commented on src/shared/api/getUser.ts:
```diff
@@ -0,0 +1,31 @@
+
+  if (!response.ok) {
+    const error = await response.json();
+    throw new Error(error.message || "Something went wrong!");
```
пользователь тоже хочет видеть ошибки, предлагаю добавить нотификашку какую-то

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/a7be9fc99082854d6edfe1404979c3bb156be87d

remadyt commented on src/shared/api/getUser.ts:
```diff
@@ -0,0 +1,31 @@
+import type { User } from "../types";
+
+const getUser = async ({ query }: { query: string }): Promise<User> => {
+  const response = await fetch(`https://api.github.com/users/${query}`);
```
думаю стоит написать функцию фетчер, которую можно будет переиспользовать и не дублировать код

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/33d3d4ab3174136880172b16efdec6e1e312a944

remadyt commented on src/shared/api/useService.ts:
```diff
@@ -0,0 +1,54 @@
+    };
+
+    if (!loading) { fetchData(); }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
не нужно выключать ошибки еслинта

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/d94a411ba36c617de639c1efec64f04cc87f914e

remadyt commented on src/shared/api/useService.ts:
```diff
@@ -0,0 +1,54 @@
+    }
+    const fetchData = async (): Promise<void> => {
+      setLoading(true);
+      setError(undefined);
```
для чего? дефолтное значение и так undefined. вместо undefined лучше юзать пустую строку или налл

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/33d3d4ab3174136880172b16efdec6e1e312a944

remadyt commented on src/shared/api/useService.ts:
```diff
@@ -0,0 +1,54 @@
+        setData(fetchedData);
+      } catch (e) {
+        setError(e as Error);
+        setData(null);
```
тоже зачем? дефолтное значение и так налл

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/commit/33d3d4ab3174136880172b16efdec6e1e312a944

remadyt commented on src/shared/api/useService.ts:
```diff
@@ -0,0 +1,54 @@
+
+  useEffect(() => {
+    if (loading) { return; }
+    if (query.length === 0 && error) {
```
нельзя обойтись без этих 3 ифов?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/33d3d4ab3174136880172b16efdec6e1e312a944

remadyt commented on src/shared/api/useService.ts:
```diff
@@ -0,0 +1,54 @@
+  useEffect(() => {
+    if (loading) { return; }
+    if (query.length === 0 && error) {
+      setError(undefined);
```
нужно вынести эту всю лоигку из юзфеекта

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/33d3d4ab3174136880172b16efdec6e1e312a944

remadyt commented on src/shared/types/index.ts:
```diff
@@ -0,0 +1,46 @@
+  error: Error | undefined;
+}
+
+type PageProps = {
```
пропсы лучше держать в компоненте

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/731dff43567b0452ec9cdd737d4c4229d848c629

remadyt commented on src/shared/types/index.ts:
```diff
@@ -0,0 +1,46 @@
+
+type ServicePayload<T> = {
+  loading: boolean,
+  error: Error | undefined,
```
если хочешь сделать поле не обязательным, то: ```error?: Error```

remadyt commented on src/shared/ui/search-input/index.tsx:
```diff
@@ -0,0 +1,26 @@
+		<input
+			className={css.search__input}
+			{...props}
+			onChange={(e: ChangeEvent<HTMLInputElement>): void => {
```
нужно выносить из jsx, и лушче юзать currentTarget

https://www.google.com/search?q=currentTarget+vs+target&oq=currentTarget+vs+target&gs_lcrp=EgZjaHJvbWUyCQgAEEUYORiABDIHCAEQABiABDIHCAIQABiABDIICAMQABgWGB4yCAgEEAAYFhgeMggIBRAAGBYYHjIICAYQABgWGB4yCAgHEAAYFhgeMggICBAAGBYYHjIICAkQABgWGB7SAQg1MjAxajBqN6gCALACAA&sourceid=chrome&ie=UTF-8

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c53fc7cf36c2f9781e722cd5b29b3355640940a4

remadyt commented on src/shared/ui/svg/index.tsx:
```diff
@@ -0,0 +1,158 @@
+const SVGSelector = ({ id }: { id: string }): JSX.Element => {
+  switch (id) {
```
попробуй вот так

import { ReactComponent as Icon } from './icon.svg';

<Icon />

svgr плагин посмотри

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/facacd433fd49e01093780458ea44479d3597ade

remadyt commented on src/widgets/repo-list/api/getRepos.ts:
```diff
@@ -0,0 +1,18 @@
+import type { Repo } from "../../../shared/types";
```
поправь путь,должен быть абсолютным

remadyt commented on src/widgets/repo-list/ui/pagination/index.tsx:
```diff
@@ -0,0 +1,83 @@
+								!cur.hidden ||
+								(index > 0 && !options[index - 1]?.hidden) ||
+								index === 0
+						)
```
преттиер работает? очень трудно это читать

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/d94a411ba36c617de639c1efec64f04cc87f914e

milabozhevskaya committed:
chore(project): configure hasky, commitizen, commitlint, prettier

milabozhevskaya committed:
refactor(project): replace fetch response with tanstack and axios

milabozhevskaya committed:
refactor(shared/ui/svg): implement svgr fot svg

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578324414

milabozhevskaya committed:
refactor(project): replace css to styles, del return type JSX, remove wrapper user-info

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578310826,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578311751,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578312263

milabozhevskaya committed:
refactor(project): implement mixin font with global font settings

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578312755,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578313325,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578313449

milabozhevskaya committed:
refactor(project): move all collors to variable file

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578314271

milabozhevskaya committed:
refactor(shared/api/services, shared/api/constants): move base url to constants file

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578315688

milabozhevskaya committed:
refactor(project): refactor type declarations

n

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578321838,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578321179

milabozhevskaya committed:
refactor(search, search-input): del props, add onChange input handler

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578323266,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578313930

milabozhevskaya committed:
feat(project): implement error notification

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1578316261

milabozhevskaya committed:
refactor(project): rename classes from snakeCase style to camelCase style

https://github.com/milabozhevskaya/fronttech/pull/1

milabozhevskaya committed:
refactor(project): implement import entities from React

https://github.com/milabozhevskaya/fronttech/pull/1

suveiko commented on src/app/app.tsx:
```diff
@@ -0,0 +1,61 @@
+    userQuery
+  );
+
+  useEffect(() => {
```
обработку ошибки можно сделать без useEffect, завяжись на onError в хуке запроса

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/ff32a6e3189a460d571f66b57a8b42c79ae04d5c

suveiko commented on src/app/app.tsx:
```diff
@@ -0,0 +1,61 @@
+import "./styles/styles.scss";
+
+const App = () => {
+  const [userQuery, setUserQuery] = useState("");
```
userQuery? нейминг сбивает с толку, это значение инпута?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/ff32a6e3189a460d571f66b57a8b42c79ae04d5c

suveiko commented on src/app/app.tsx:
```diff
@@ -0,0 +1,61 @@
+        userQuery={userQuery}
+        loading={isLoadingUser}
+        fetching={isFetchingUser}
+        notFound={!!errorUser}
```
от этих пропсов можно избавиться, почему не хочешь вынести все в контенкст?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/ff32a6e3189a460d571f66b57a8b42c79ae04d5c

suveiko commented on src/entities/card/styles.module.scss:
```diff
@@ -0,0 +1,21 @@
+  cursor: pointer;
+}
+
+.card-description {
```
нейминг должен быть везде одинаковый, используй везде camelCase

milabozhevskaya replied:
название селекторов в camelCase ?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/58990bc34c5d6bf904bbf77ca8941cf7efeb5d01

suveiko commented on src/entities/user/index.tsx:
```diff
@@ -0,0 +1,47 @@
+}: Exclude<User, "publicRepos">) => (
+  <>
+    <Avatar>
+      <img src={avatarURL} alt={name} />
```
img нельзя закинуть внутрь Avatar?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/f2feb042d633d6dbc86254d93b7d9a5e24b9cef1

suveiko commented on src/shared/api/services.ts:
```diff
@@ -0,0 +1,15 @@
+    axios.get<User>(`${BASE_URL}/users/${user}`),
+
+  getRepos: async ({ user, perPage, page }: ReposQuery) =>
+    axios.get<Array<Repo>>(
```
создай axios instance

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/be36c85d601070684a80361b047314e1cdeeba8f

suveiko commented on src/shared/constants/index.ts:
```diff
@@ -0,0 +1,13 @@
+const BASE_URL = "https://api.github.com";
+const QUERY_KEY_USER = "user";
```
это излишне

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/3b2f4288705959602d1bbeda428947e6e8312028

suveiko commented on src/shared/constants/index.ts:
```diff
@@ -0,0 +1,13 @@
+const BASE_URL = "https://api.github.com";
+const QUERY_KEY_USER = "user";
+const QUERY_KEY_REPOS = "repos";
```
это излишне

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/3b2f4288705959602d1bbeda428947e6e8312028

suveiko commented on src/shared/types/index.ts:
```diff
@@ -0,0 +1,56 @@
+  : T extends Array<Repo>
+    ? ReposQuery
+    : ReposQuery;
+// : Record<string, string>;
```
удали закоменч код

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c8ffc0d9423399d0f68088cb2719b3e8a62266fc

suveiko commented on src/shared/ui/svg/arrow-left.source.svg:
можно обойтись 1 стрелочкой, и переворачивать ее стилями

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/76131171dd1555e38f27ca2771f23ee30e384f1d

suveiko commented on src/widgets/header/index.tsx:
```diff
@@ -0,0 +1,12 @@
+import { Logo } from "@/shared/ui/logo";
+import { SearchInput } from "@/shared/ui/search-input";
+
+const Header = ({ onSearch }: { onSearch: (value: string) => void }) => (
```
создай отдельный тип Props здесь и во всех похожих местах

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c8ffc0d9423399d0f68088cb2719b3e8a62266fc

suveiko commented on src/widgets/repo-list/ui/pagination/index.tsx:
```diff
@@ -0,0 +1,83 @@
+        <button
+          className={styles.paginationButtonNav}
+          onClick={(): void => {
+            setPage(page - 1);
```
вынеси в отдельную функцию

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/622d4b82a0eb6509f3e46095a5261e81d653d87b

suveiko commented on src/widgets/repo-list/ui/pagination/index.tsx:
```diff
@@ -0,0 +1,83 @@
+                <button
+                  key={value}
+                  className={`${styles.paginationButton} ${styles.paginationButton_dots}`}
+                  onClick={(): void => {}}
```
так нужно?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/622d4b82a0eb6509f3e46095a5261e81d653d87b

suveiko commented on src/widgets/repo-list/ui/pagination/index.tsx:
```diff
@@ -0,0 +1,83 @@
+                  <button
+                    key={value}
+                    className={styles.paginationButton}
+                    onClick={(): void => {
```
вынеси в отедельную функцию, и в :void нет особого смысла, typescript делает это автоматически, явно указывать тип не нужно, если это требует eslint, нужно выкличить это правильно, либо доработать

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/622d4b82a0eb6509f3e46095a5261e81d653d87b

suveiko commented on src/entities/card/index.tsx:
```diff
@@ -0,0 +1,19 @@
+
+import styles from "./styles.module.scss";
+
+const Card = ({ name, description, html_url: htmlURL }: Repo) => (
```
```suggestion
const Card: FC<Repo> = ({ name, description, html_url: htmlURL }) => (
```

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c8ffc0d9423399d0f68088cb2719b3e8a62266fc

suveiko commented on src/entities/user/index.tsx:
```diff
@@ -0,0 +1,47 @@
+
+import styles from "./styles.module.scss";
+
+const UserInfo = ({
+  name,
+  login,
+  avatar_url: avatarURL,
+  html_url: htmlURL,
+  followers,
+  following,
+}: Exclude<User, "publicRepos">) => (
```
```suggestion
const UserInfo: FC<Exclude<User, "publicRepos">> = ({
  name,
  login,
  avatar_url: avatarURL,
  html_url: htmlURL,
  followers,
  following,
}) => (
```

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c8ffc0d9423399d0f68088cb2719b3e8a62266fc

suveiko commented on src/entities/user/index.tsx:
```diff
@@ -0,0 +1,47 @@
+  </>
+);
+
+export { UserInfo };
```
почему сразу не написать export const UserInfo?

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/2ce7dea13d657bbde87f7c8c6df63f29df8e6240

suveiko commented on src/pages/index.tsx:
```diff
@@ -0,0 +1,28 @@
+  notFound: boolean;
+};
+
+const Page = ({ userQuery, user, loading, fetching, notFound }: PageProps) => {
+  if (loading || fetching) {
+    return <LoadingPage />;
+  }
+  if (!loading && notFound && !user) {
+    return <NotFoundPage />;
+  }
+  if (!loading && !notFound && user) {
+    return <UserPage {...user} userQuery={userQuery} />;
+  }
+  return <EmptyPage />;
+};
+
```
можно попробовать на switch переписать, но вроде и так нормально

suveiko commented on src/pages/user-page/index.tsx:
```diff
@@ -0,0 +1,24 @@
+
+type UserPageProps = User & { userQuery: string };
+
+const UserPage = ({ ...props }: UserPageProps) => (
```
```suggestion
const UserPage: FC<UserPageProps> = ({ ...props }) => (
```

milabozhevskaya replied:
https://github.com/milabozhevskaya/fronttech/pull/1/commits/c8ffc0d9423399d0f68088cb2719b3e8a62266fc

milabozhevskaya committed:
refactor(shared/ui/avatar, entities/user): change Avatar component structure

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584427354

milabozhevskaya committed:
refactor(project): del service hook, change structure request, add context

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584423209,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584424503,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584425990

milabozhevskaya committed:
fix(pages, repo-list, widgets/list): bug with show loading list

milabozhevskaya committed:
refactor(services): implement instans of axios

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584429244

milabozhevskaya committed:
refactor(shared/ui/svg, widgets/repo-list/pagination): using arrow-left instead arrow-right

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584432771

milabozhevskaya committed:
refactor(project): implement using module exports with module definitions

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584483500

milabozhevskaya committed:
refactor(project): implement context for page number, add button click  handler

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584435672,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584436538,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584438365

milabozhevskaya committed:
refactor(shared/constants): remove unnecessary constants

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584430186,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584430324

milabozhevskaya committed:
refactor(project): integrate type declarations with module

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584430738,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584433754,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584481253,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584482401,
https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584486275

milabozhevskaya committed:
refactor(project): change style of select naming

https://github.com/milabozhevskaya/fronttech/pull/1#discussion_r1584426716
